### PR TITLE
C API: Add BinaryenFunctionAppendVar

### PR DESCRIPTION
**Pull request by mobsceneZ**
Refer to issue #6214, this pull request is trying to allow Binaryen C API to append new local variables to the function (in cases that there does not exist such C API yet).

### CODE REVIEW

File: src/binaryen-c.h [hunk 1]
```diff
@@ -3197,6 +3197,10 @@ BINARYEN_API BinaryenIndex BinaryenFunctionGetNumVars(BinaryenFunctionRef func);
 // specified `Function`.
 BINARYEN_API BinaryenType BinaryenFunctionGetVar(BinaryenFunctionRef func,
                                                  BinaryenIndex index);
+// Appends a local variable to the specified `Function`, returning its
+// insertion index.
```
Comment (kripken) [inline src/binaryen-c.h:3201]:
```suggestion
// index.
```

File: src/binaryen-c.cpp [hunk 2]
```diff
@@ -5878,6 +5878,14 @@ BinaryenType BinaryenFunctionGetVar(BinaryenFunctionRef func,
   assert(index < vars.size());
   return vars[index].getID();
 }
+BinaryenIndex BinaryenFunctionAppendVar(BinaryenFunctionRef func,
+                                        BinaryenType type) {
+  auto* function = (Function*)func;
+  auto& list = function->vars;
+  auto index = list.size();
+  list.push_back((Type)type);
```
Comment (kripken) [inline src/binaryen-c.cpp:5886]: This can use `Builder::addVar`.

For consistency with that API, perhaps this could be `BinaryenFunctionAddVar`?

**Review by kripken — COMMENTED** [review]
Looks like a reasonable API to add.

See comments, + also add a tiny test somewhere in `test/example/c-api-kitchen-sink.c`.

**Comment by mobsceneZ** [comment]
> Looks like a reasonable API to add.
> 
> See comments, + also add a tiny test somewhere in `test/example/c-api-kitchen-sink.c`.

Hey kripken, according to your suggestions I made corresponding changes which can be seen in commit 7e968bdf34ba3ac3ed9d2ea2f198d654bd4fce30. You can check that out to see if it's what you want ;).

File: test/example/c-api-kitchen-sink.c [hunk 3]
```diff
@@ -1417,6 +1417,12 @@ void test_core() {
   BinaryenFunctionRef sinker = BinaryenAddFunction(
     module, "kitchen()sinker", iIfF, BinaryenTypeInt32(), localTypes, 2, body);
 
+  BinaryenIndex numLocals = BinaryenFunctionGetNumLocals(sinker);
+  BinaryenIndex numParams = BinaryenTypeArity(BinaryenFunctionGetParams(sinker));
+  BinaryenIndex newLocalIdx = BinaryenFunctionAddVar(sinker, BinaryenTypeFloat32());
+  assert(newLocalIdx == numLocals);
+  assert(BinaryenFunctionGetVar(sinker, newLocalIdx - numParams) == BinaryenTypeFloat32());
+
```
Comment (kripken) [inline test/example/c-api-kitchen-sink.c:1425]: We can also check the number of vars was incremented properly.
```suggestion
  assert(BinaryenFunctionGetNumLocals(sinker) == numLocals + 1);

```

**Comment by kripken** [comment]
See also the error on CI about some small lint changes that are required.

**Comment by mobsceneZ** [comment]
> See also the error on CI about some small lint changes that are required.

Hey Kripken, I made corresponding changes in commit b2a69dda6269491c8e3261693b9beabfa799a24a (it's a rewrite of the last commit), you can check that out to see if that has further problems.

**Review by kripken — APPROVED** [review]

**Pull request merged**